syntax = "proto3";
//import "google/protobuf/timestamp.proto";

message UrpProduct {
  string request_id = 1;
  Transaction transaction = 2;
}

message Transaction {
  string trusted_system = 2;
  repeated UrpObject new_objects = 3;
  repeated UrpObject update_objects = 4;
  repeated Relationship relationships = 5;
  string date_time = 6;
  //  google.protobuf.Timestamp date_time = 5;
}
message UrpObject {
  string urpid = 1;
  repeated string aliases = 2;
  string object_type = 3;
  string object_subtype = 4;
  repeated Alternative_id alternative_ids = 5;
}

message Alternative_id {
  string alternative_id = 1;
  CodingSystem coding_system = 2;
}

message Relationship {
  string relationship_type = 1;
  string parent_object_urpid = 2;
  string child_object_urpid = 3;
}

message CodingSystem {
  string coding_system_code = 1;
  string url_prefix = 2;
}




